python-checks/src/test/resources/checks/verifiedSslTlsCertificateCheck.py

```diff
@@ -0,0 +1,282 @@
+def pyopensslTest():
+  # Mutably borrowed from here:
+  # https://github.com/SonarSource/security-expected-issues/blob/master/python/rules/vulnerabilities/\
+  # RSPEC-4830%20Server%20certificates%20should%20be%20verified%20during%20SSL%E2%81%84TLS%20connections/\
+  # pyopenssl-test.py
+
+  from OpenSSL import SSL
+  import sys, os, select, socket
+
+  # Valid modes are VERIFY_NONE, VERIFY_PEER, VERIFY_CLIENT_ONCE, VERIFY_FAIL_IF_NO_PEER_CERT and defined on OpenSSL::SSL
+  # The default mode is VERIFY_NONE, which does not perform any verification at al
+  #
+  # VERIFY_PEER Don't check peer certificate.
+  # VERIFY_FAIL_IF_NO_PEER_CERT Client:Check peer certificate, Server:Check peer certificate and skip if no certificate received.
+  #  If VERIFY_PEER is used, mode can be OR:ed with VERIFY_FAIL_IF_NO_PEER_CERT and VERIFY_CLIENT_ONCE to further control the behaviour.
+
+  ctx1 = SSL.Context(SSL.TLSv1_2_METHOD)
+  ctx1.set_verify(SSL.VERIFY_PEER, verify_callback) # Compliant
+  ctx1.set_verify(SSL.VERIFY_PEER | SSL.VERIFY_FAIL_IF_NO_PEER_CERT, verify_callback) # Compliant
+  ctx1.set_verify(SSL.VERIFY_PEER | SSL.VERIFY_FAIL_IF_NO_PEER_CERT | VERIFY_CLIENT_ONCE, verify_callback) # Compliant
+
+  def verify_callback(connection, x509, errnum, errdepth, ok):
```

Comment:
I'm not sure we really need to have all this level of detail in unit tests, maybe an empty function would be ok